repos/system_upgrade/common/actors/systemfacts/libraries/systemfacts.py

```diff
@@ -289,7 +289,9 @@ def _default_grub_info():
         ])
     else:
         for line in run(['cat', default_grb_fpath], split=True)['stdout']:
-            if not line.strip():
+            line = line.strip()
+            if not line or line[0] == '#':
+                # skip comments and empty lines
                 continue
             name, value = tuple(map(type(line).strip, line.split('=', 1)))
```

Comment:
```suggestion
            line = line.strip()
            if not line or line[0] == '#':
                # skip comments and empty lines
                continue
            try:
                name, value = tuple(map(type(line).strip, line.split('=', 1)))
            except ValueError:
                continue 
```

Reply:
@vinzenz I do not think this is what we want. From the point that in case we are not able to parse the file, we are not sure it's valid (we hit already issues with invalid default grub file long time ago). From this point, I think we should not allow people to continue. that's why I choosed this solution. But I think we should raise correct error msg to point out what lines are problematic so we can decide whether the problem is on our or customers side when we get the report

Reply:
discussed over irc. we will go with the strict parsing right now. discussion is opened in future